### PR TITLE
LEP-345 move remark handling up call stack

### PR DESCRIPTION
**Pull request by starswan**
https://dsdmoj.atlassian.net/browse/LEP-345

In order to run the pipeline multiple times, the remarks side-effects have to be removed out of the core and into the higher levels of the code.

---

## Checklists

Author: (before you ask people to review this PR)

- [ ] Diff - review it, ensuring it contains only expected changes
- [ ] Whitespace changes - avoid if possible, because they make diffs harder to read and conflicts more likely
- [ ] Changelog - add a line, if it meets the criteria
- [ ] Secrets - no secrets should be added
- [ ] Commit messages - say *why* the change was made
- [ ] PR description - summarizes *what* changed and *why*, with a JIRA ticket ID
- [ ] Tests pass - on CircleCI
- [ ] Conflicts - resolve if Github reports them. e.g. with `git rebase main`

Reviewers remember:

- [ ] Jira ticket criteria are met
- [ ] Migrations - test migration and rollback: `rake db:migrate && rake db:rollback`

### CODE REVIEW

File: app/controllers/v6/assessments_controller.rb [hunk 1]
```diff
@@ -27,33 +27,31 @@ def create
                                 applicant_outgoings)
 
         partner_params = full_assessment_params[:partner]
-        if partner_params.present?
-          partner_dependants = dependants partner_params, create.assessment.submission_date
-          render_unprocessable(dependant_errors(partner_dependants)) && return if partner_dependants.reject(&:valid?).any?
+        calculation_output = if partner_params.present?
```
Comment (ripan) [inline app/controllers/v6/assessments_controller.rb:30]: @starswan I thought of removing else by  assigning partner to nil before condition. What do you think?

File: app/services/workflows/main_workflow.rb [hunk 2]
```diff
@@ -2,34 +2,62 @@ module Workflows
...
+
+          remarks = RemarkGenerators::Orchestrator.call(employments: applicant.employments,
+                                                        gross_income_summary: assessment.applicant_gross_income_summary,
+                                                        outgoings: applicant.outgoings,
+                                                        liquid_capital_items: applicant.capitals_data.liquid_capital_items,
+                                                        state_benefits: applicant.state_benefits,
+                                                        lower_capital_threshold:,
+                                                        child_care_bank: calculation_output.applicant_disposable_income_subtotals.child_care_bank,
+                                                        assessed_capital:)
+          assessment.add_remarks!(remarks)
```
Comment (ripan) [inline app/services/workflows/main_workflow.rb:49]: @starswan We were concating remarks before and made single insert. Now we have 2 inserts for applicant and partner. Would it create any issue?

File: app/services/workflows/non_passported_workflow.rb [hunk 3]
```diff
@@ -1,37 +1,39 @@
...
+                                 capital_subtotals = get_capital_subtotals(assessment:, applicant:, partner:, disposable_income_subtotals:)
+                                 CalculationOutput.new(gross_income_subtotals: gross_income_subtotals.gross,
+                                                       disposable_income_subtotals:, capital_subtotals:,
+                                                       proceeding_types: assessment.proceeding_types,
+                                                       receives_qualifying_benefit: applicant.details.receives_qualifying_benefit,
+                                                       receives_asylum_support: applicant.details.receives_asylum_support)
+                               end
+                             end
+
+        Result.new calculation_output:, remarks: gross_income_subtotals.remarks
```
Comment (ripan) [inline app/services/workflows/non_passported_workflow.rb:36]: @starswan What do you think of returning similar Result object for Passported workflow?

Comment (starswan) [reply app/services/workflows/non_passported_workflow.rb:36]: I thought of that, but it would always be empty.

**Review by ripan — APPROVED** [review]

**Pull request merged**